pkg/services/rpcsrv/server.go

```diff
@@ -153,6 +153,9 @@ type (
 		notificationCh  chan *state.ContainedNotificationEvent
 		transactionCh   chan *transaction.Transaction
 		notaryRequestCh chan mempoolevent.Event
+
+		// optional function called after Shutdown completion
+		onShutdown func()
```

Comment:
Just for the test?

Reply:
yep, within test I need to check that async `Shutdown` finished. Maybe we'll find more beautiful solution with sync `Shutdown` https://github.com/nspcc-dev/neo-go/pull/2966#discussion_r1165474594  and this won't be needed

Reply:
Move this to the test side, there's no need to have it inside the server. Just make this inside the test:
```
go func(){
    srv.Shutdown()
    onShutdown()
}()
```
This will solve the issue.

Reply:
done